src/gleam_stdlib.mjs

```diff
@@ -612,6 +612,49 @@ export function decode_tuple(data) {
   return Array.isArray(data) ? new Ok(data) : decoder_error("Tuple", data);
 }
 
+export function decode_tuple2(data) {
+  return decode_tupleN(data, 2);
+}
+
+export function decode_tuple3(data) {
+  return decode_tupleN(data, 3);
+}
+
+export function decode_tuple4(data) {
+  return decode_tupleN(data, 4);
+}
+
+export function decode_tuple5(data) {
+  return decode_tupleN(data, 5);
+}
+
+export function decode_tuple6(data) {
+  return decode_tupleN(data, 6);
+}
+
+function decode_tupleN(data, n) {
+  let error_message = `Tuple or List of ${n} elements`;
+
+  if (!(List.isList(data) || Array.isArray(data))) {
+    return decoder_error(error_message, data);
+  }
+
+  let array = List.isList(data) ? [...data] : data; 
```

Comment:
This traverses the full list, we only want to check the length up to the desired length, failing if it is too long.

Something like this:

```js
function decode_tupleN(data, n) {
  if (Array.isArray(data) && data.length == n) {
    return new Ok(data)
  }

  let list = decode_exact_length_list(data, n)
  if (list) return new Ok(list)

  return decoder_error(`Tuple of ${n} elements`, data);
}

function decode_exact_length_list(data, n) {
  if (!List.isList(data)) return;

  let elements = []
  let current = data

  for (let i = 0; i < n; i++) {
    if (current.isEmpty()) break;
    element.push(current.head)
    current = current.tail
  }

  if (elements.length === n) return elements.reverse()
}
```